backend/curriculum_tracking/api_views.py

```diff
@@ -883,6 +891,26 @@ def auto_assign_reviewers(self, request, pk=None):
             response = actor.send()
             return Response({"status": "OK", "data": response.asdict()})
 
+    @action(
+        detail=False,
+        methods=["POST"],
+        serializer_class=serializers.SetDueTimeSerializer,
+        permission_classes=[HasObjectPermission(permissions=Team.PERMISSION_MANAGE_CARDS)]
+    )
+    def bulk_set_due_dates(self, request, pk=None):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            teams = self.get_object()   # This will get you all the teams, we only need the teams in the request data
+            for team in teams:
+                if team.name in request.data.get('team'):
+                    team_cards = get_team_cards(team, request.data.get('content_item'))
```

Comment:
Should we worry about flavour_matching?

Reply:
I don't think so, take for instance the team provisionally accepted java, each team member will only be doing java, so the cards we get from calling 'get_team_cards' should therefore only return 'java cards'.